cmd/kubectl-gadget/version.go

```diff
@@ -15,16 +15,29 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
 
-	"github.com/spf13/cobra"
-
 	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/utils"
 	"github.com/inspektor-gadget/inspektor-gadget/internal/version"
 	grpcruntime "github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/grpc"
+	"github.com/spf13/cobra"
 )
 
+// VersionInfo represents the structure for version information output
+type VersionInfo struct {
+	ClientVersion *Version `json:"clientVersion,omitempty"`
+	ServerVersion *Version `json:"serverVersion,omitempty"`
+}
+
+// Version contains detailed version information
+type Version struct {
+	Version string `json:"version"`
+	Status  string `json:"status,omitempty"`
+}
+
 func init() {
+	versionCmd.Flags().StringP("output", "o", "", "Output format. One of: json|''")
```

Comment:
nit: I think it's better to use StringVarP, so you have a global variable with the value and don't need to call `cmd.Flags().GetString("output")` below.

Reply:
I agree StringVarP would make the code cleaner